prod/prod.go

```diff
@@ -18,44 +18,95 @@ package prod
+var ErrPubKey = errors.New("failed to decompress public key")
+
+// ErrFeedLookup is returned when the recovery feed cannot be successefully looked up
+var ErrFeedLookup = errors.New("failed to look up recovery feed")
+
+// ErrFeedContent is returned when there is a failure to retrieve content from the recovery feed
+var ErrFeedContent = errors.New("failed to get content for recovery feed")
+
+// RecoveryHook defines code to be executed upon failing to retrieve pinned chunks
+type RecoveryHook func(ctx context.Context, chunkAddress chunk.Address, publisher string) error
+
+// sender is the function type for sending trojan chunks
 type sender func(ctx context.Context, targets [][]byte, topic trojan.Topic, payload []byte) (*pss.Monitor, error)
 
 // NewRecoveryHook returns a new RecoveryHook with the sender function defined
-func NewRecoveryHook(send sender) RecoveryHook {
-	return func(ctx context.Context, chunkAddress chunk.Address) error {
-		targets, err := getPinners(chunkAddress)
+func NewRecoveryHook(send sender, handler feed.GenericHandler) RecoveryHook {
+	return func(ctx context.Context, chunkAddress chunk.Address, publisher string) error {
+		targets, err := getPinners(publisher, handler)
 		if err != nil {
 			return err
 		}
 		payload := chunkAddress
-		topic := trojan.NewTopic("RECOVERY")
 
 		// TODO: monitor return should
-		if _, err := send(ctx, targets, topic, payload); err != nil {
+		if _, err := send(ctx, targets, trojan.RecoveryTopic, payload); err != nil {
 			return err
 		}
 		return nil
 	}
 }
 
-// TODO: refactor this method to implement feed of target pinners
-// getPinners returns the specific target pinners for a corresponding chunk address
-func getPinners(chunkAddress chunk.Address) ([][]byte, error) {
-	//this should get the feed and return correct target of pinners
-	return [][]byte{
-		{57, 120},
-		{209, 156},
-		{156, 38},
-		{89, 19},
-		{22, 129}}, nil
+// getPinners returns the specific target pinners for a corresponding chunk
+func getPinners(publisher string, handler feed.GenericHandler) ([][]byte, error) {
+	// get feed user from publisher
+	publisherBytes, err := hex.DecodeString(publisher)
+	if err != nil {
+		return nil, ErrPublisher
+	}
+	pubKey, err := crypto.DecompressPubkey(publisherBytes)
+	if err != nil {
+		return nil, ErrPubKey
+	}
+	addr := crypto.PubkeyToAddress(*pubKey)
+
+	// get feed topic from trojan recovery topic
+	topic, err := feed.NewTopic(trojan.RecoveryTopicText, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// read feed
+	fd := feed.Feed{
+		Topic: topic,
+		User:  addr,
+	}
+
+	query := feed.NewQueryLatest(&fd, lookup.NoClue)
+	// TODO: do we need WithCancel?
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	_, err = handler.Lookup(ctx, query)
+	// feed can still be queried even if there are no updates
+	if err != nil && err.Error() != "no feed updates found" {
```

Comment:
i would much rather make the comparison here to `error`-typed consts or vars, but this is the way errors are handled in the `feeds` package